File: specs/fixtures/sources/multi_mutation.rs
// Multiple mutations within a single operation.
// The system must capture every mutation of count, not just boundaries.
use specgate_annotations::*;

#[spec_setup("make_counter")]
fn make_counter() -> Counter {
    Counter { count: 0 }
}

struct Counter {
    #[spec_event]
    count: i32,
}

impl Counter {
    #[spec_operation("increment_twice")]
    fn increment_twice(&mut self) {
        self.count += 1;
        self.count += 1;
    }
}

File: test/csharp/SpecGateFixtures.Net8/Conformance/Stateful/MultiMutation/multi_mutation.cs
using SpecGate.Annotations;
namespace SpecGateFixtures.Conformance.Stateful.MultiMutation;

/// <summary>
/// net8.0 copy of the multi-mutation counter fixture — the cross-framework
/// witness proving intermediate-mutation capture works on an older TFM.
/// </summary>
public class Counter
{
    /// <summary>The current count; each assignment is captured as a distinct mutation.</summary>
    [SpecEvent("count")]
    public int Count { get; set; }

    /// <summary>Builds a counter starting at zero.</summary>
    /// <returns>A new <see cref="Counter"/> with <see cref="Count"/> 0.</returns>
    [SpecSetup("increment_twice")]
    public static Counter Make() => new() { Count = 0 };

    /// <summary>Increments the counter twice, emitting both intermediate values.</summary>
    [SpecOperation("increment_twice")]
    public void IncrementTwice()
    {
        Count += 1;
        Count += 1;
    }
}
